network-interface/src/network.rs

```diff
@@ -157,16 +158,18 @@ pub trait Network: Send + Sync + Unpin + 'static {
         T: Topic + Sync;
 
     /// Gets a value from the distributed hash table
-    async fn dht_get<K, V>(&self, k: &K) -> Result<Option<V>, Self::Error>
+    async fn dht_get<K, V, T>(&self, k: &K) -> Result<Option<V>, Self::Error>
     where
         K: AsRef<[u8]> + Send + Sync,
-        V: Deserialize + Send + Sync;
+        V: Deserialize + Send + Sync + TaggedSignable + PartialOrd,
```

Comment:
```suggestion
        V: Deserialize + Send + Sync + TaggedSignable + Ord,
```
`Ord` would be more appropriate because it implies stuff like `a == a` and `a <= b && b <= c` implying `a <= c`. See https://doc.rust-lang.org/1.75.0/std/cmp/trait.Ord.html#corollaries.